test/e2e/mv3/multiple-restarts.spec.js

```diff
@@ -0,0 +1,294 @@
+  it('Simple simple send flow within full screen view should still be usable', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions: generateGanacheOptions(),
+        title: this.test.title,
+        driverOptions: { openDevToolsForTabs: true },
+      },
+      async ({ driver, ganacheServer }) => {
+        await driver.navigate();
+
+        await unlockWallet(driver, WALLET_PASSWORD);
+
+        await assertETHBalance(ganacheServer, '25');
+
+        console.log('first send ETH and then terminate SW');
+        const RECIPIENT_ADDRESS = '0x985c30949c92df7a0bd42e0f3e3d539ece98db24';
+        await simpleSendETH(driver, '1', RECIPIENT_ADDRESS);
+        await terminateServiceWorker(driver);
+
+        await assertETHBalance(ganacheServer, '24.0000');
+
+        console.log('first send ETH #2 and then terminate SW');
+        await switchToWindow(driver, WINDOW_TITLES.ExtensionInFullScreenView);
+        await simpleSendETH(driver, '1', RECIPIENT_ADDRESS);
+        await terminateServiceWorker(driver);
+
+        await assertETHBalance(ganacheServer, '22.9999');
+
+        console.log('first terminate SW and then send ETH');
+        await switchToWindow(driver, WINDOW_TITLES.ExtensionInFullScreenView);
+        await terminateServiceWorker(driver);
+
+        await switchToWindow(driver, WINDOW_TITLES.ExtensionInFullScreenView);
+        await simpleSendETH(driver, '1', RECIPIENT_ADDRESS);
+
+        await sleepSeconds(2);
+        await assertETHBalance(ganacheServer, '21.9999');
+      },
+    );
+  });
+
+  it('Should continue to support dapp interactions after service worker re-starts multiple times', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions: generateGanacheOptions({
+          concurrent: { port: 8546, chainId: 1338 },
+        }),
+        title: this.test.title,
+        driverOptions: { openDevToolsForTabs: true },
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+
+        await unlockWallet(driver, WALLET_PASSWORD);
+
+        await openDapp(driver);
+
+        console.log('Click add Ethereum chain');
+        await switchToWindow(driver, WINDOW_TITLES.TestDApp);
+        await driver.clickElement('#addEthereumChain');
+        await driver.waitUntilXWindowHandles(2);
+
+        console.log('Notification pop up opens');
+        await switchToWindow(driver, WINDOW_TITLES.Notification);
+        let notification = await driver.isElementPresent({
+          text: 'Allow this site to add a network?',
+          tag: 'h3',
+        });
+        assert.ok(notification, 'Dapp action does not appear in Metamask');
+
+        console.log('Cancel Notification');
+        await driver.clickElement({ text: 'Cancel', tag: 'button' });
```

Comment:
we can also skip the cancel part, and just accumulate the txns in the end. This would test the persistent old txn and accept new ones, would this also be helpful?